lisa/tools/lagscope.py

```diff
@@ -0,0 +1,258 @@
+    #     345          141943
+    #     360          62452
+    #     375          32662
+    #     390          10980
+    #     405          3605
+    #     420          1786
+    #     435          992
+    #     450          639
+    #     465          457
+    #     480          2204
+    _interval_frequency_pattern = re.compile(
+        r"\s+(?P<interval_us>\d+)\s+(?P<frequency>\d+)$", re.M
+    )
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Git]
+
+    @property
+    def command(self) -> str:
+        return "lagscope"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def _install(self) -> bool:
+        self._install_dep_packages()
+        tool_path = self.get_tool_path()
+        git = self.node.tools[Git]
+        git.clone(self.repo, tool_path, ref=self.branch)
+        code_path = tool_path.joinpath("lagscope")
+        self.node.execute(
+            "./do-cmake.sh build",
+            cwd=code_path,
+            shell=True,
+            expected_exit_code=0,
+            expected_exit_code_failure_message="fail to run do-cmake.sh build",
+        )
+        self.node.execute(
+            "./do-cmake.sh install",
+            cwd=code_path,
+            sudo=True,
+            shell=True,
+            expected_exit_code=0,
+            expected_exit_code_failure_message="fail to run do-cmake.sh install",
+        )
+        self.node.execute(
+            "ln -sf /usr/local/bin/lagscope /usr/bin/lagscope",
+            sudo=True,
+            shell=True,
+            expected_exit_code=0,
+            expected_exit_code_failure_message="fail to create symlink to lagscope",
+        )
+        return self._check_exists()
+
+    def run_as_server(self, ip: str = "", daemon: bool = True) -> None:
+        # -r: run as a receiver
+        # -rip: run as server mode with specified ip address
+        # -D: run as a daemon
+        cmd = ""
+        if daemon:
+            cmd += " -D"
+        if ip:
+            cmd += f" -r{ip}"
+        else:
+            cmd += " -r"
+        self.run(
+            cmd,
+            force_run=True,
+            sudo=True,
+            shell=True,
+            expected_exit_code=0,
+            expected_exit_code_failure_message=f"fail to launch cmd {self.command}"
+            f"{cmd}",
+        )
+
+    def run_as_client(
+        self,
+        server_ip: str,
+        test_interval: int = 0,
+        run_time_seconds: int = 10,
+        ping_count: int = 45000,
+        print_histogram: bool = True,
+        print_percentile: bool = True,
+        histogram_1st_interval_start_value: int = 30,
+        length_of_histogram_intervals: int = 15,
+        count_of_histogram_intervals: int = 30,
+        dump_csv: bool = True,
+        daemon: bool = False,
+    ) -> List[NetworkLatencyPerformanceMessage]:
+        # -s: run as a sender
+        # -i: test interval
+        # -t: test duration [default: 60 second(s)]
```

Comment:
It looks the ping iteration is still here, not change to the -t.

Reply:
we can specify ping count by -n100, or test duration -t30, when both specified, the ping count will be ignored.

Reply:
Please specify -t10, and check if it has about 45k test results. If so, it's good to me.

Reply:
I updated them based on actual results.